examples/src/main/java/com/mapbox/navigation/examples/core/SilentWaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,377 @@
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        // This is not the most efficient way to preserve the route on a device rotation.
+        // This is here to demonstrate that this event needs to be handled in order to
+        // redraw the route line after a rotation.
+        directionRoute?.let {
+            outState.putString(Utils.PRIMARY_ROUTE_BUNDLE_KEY, it.toJson())
+        }
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
+        super.onRestoreInstanceState(savedInstanceState)
+        directionRoute = Utils.getRouteFromBundle(savedInstanceState)
+    }
+
+    override fun onOffRouteStateChanged(offRoute: Boolean) {
+        if (offRoute) {
+            Toast.makeText(this, "You're off-route", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    override fun onRerouteStateChanged(rerouteState: RerouteState) {
+        val message: String = when (rerouteState) {
+            RerouteState.Idle -> "Reroute is idle"
+            RerouteState.Interrupted -> "Reroute request interrupted"
+            is RerouteState.Failed -> "Reroute request failed: ${rerouteState.message}"
+            RerouteState.FetchingRoute -> "Reroute request is in progress"
+            RerouteState.RouteFetched -> "Reroute request is finished successful"
+        }
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+    }
+
+    private class MyLocationEngineCallback(activity: SilentWaypointsRerouteActivity) :
+        LocationEngineCallback<LocationEngineResult> {
+
+        private val activityRef = WeakReference(activity)
+
+        override fun onSuccess(result: LocationEngineResult?) {
+            result?.locations?.firstOrNull()?.let { location ->
+                activityRef.get()?.let { activity ->
+                    val locationEvent = ReplayRouteMapper.mapToUpdateLocation(0.0, location)
+                    val locationEventList = Collections.singletonList(locationEvent)
+                    activity.mapboxReplayer.pushEvents(locationEventList)
+                    activity.mapboxReplayer.play()
+                    activity.navigationMapboxMap?.updateLocation(result.lastLocation)
+                }
+            }
+        }
+
+        override fun onFailure(exception: Exception) {
+        }
+    }
+}
+
+private class CoordinatesHolder {
+    var currentLocation: Location = Location("InternalProvider")
+    var destination: Point = Point.fromLngLat(0.0, 0.0)
+}
+
+private class WaypointsController {
+    val waypoints = mutableListOf<Point>()
+
+    fun add(latLng: LatLng) {
+        waypoints.add(latLng.toPoint())
+    }
+
+    fun clear() {
+        waypoints.clear()
+    }
+
+    fun coordinates(originLocation: Location): List<Point> {
+        val coordinates = mutableListOf<Point>()
+        coordinates.add(originLocation.toPoint())
+        coordinates.addAll(waypoints)
+        return coordinates
+    }
+}
```

Comment:
Could we extract this out into a utility class so it can be reused across examples? We have the same class in `ReplayWaypointsActivity` 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/356e9d150b2e1ca5ea5b9c7222bc222316d60d23/examples/src/main/java/com/mapbox/navigation/examples/core/ReplayWaypointsActivity.kt#L314-L331

Reply:
👍